### PR TITLE
[Feature] Fixed account for camera uploads

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
App: #3166 

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

## Screenshots

| picture_uploads_complete | picture_uploads_account_dialog |
|---|---|
_____

## QA

Test plan: https://github.com/owncloud/QA/blob/master/Mobile/Android/Release_2.18/3226-Camera%20upload%20account.md

Reports:

### CODE REVIEW

File: owncloudApp/src/main/res/xml/settings_video_uploads.xml [hunk 1]
```diff
@@ -35,13 +42,13 @@
...
         app:defaultValue="NOTHING"
         app:dialogTitle="@string/prefs_camera_upload_behaviour_dialog_title"
         app:entries="@array/pref_behaviour_entries"
         app:entryValues="@array/pref_behaviour_entryValues"
+        app:iconSpaceReserved="false"
         app:key="video_uploads_behaviour"
-        app:useSimpleSummaryProvider="true"
-        app:title="@string/prefs_camera_upload_behaviour_title" />
+        app:title="@string/prefs_camera_upload_behaviour_title"
+        app:useSimpleSummaryProvider="true" />
```
Comment (JuancaG05) [inline owncloudApp/src/main/res/xml/settings_video_uploads.xml:52]: Attribute `negativeButtonText=""` not necessary here? If it isn't, maybe we can remove them from the other ListPreferences, but I would say it is

File: owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAccount.kt [hunk 2]
```diff
@@ -27,7 +27,21 @@ const val OC_ACCOUNT_NAME = "$OC_ACCOUNT_ID@demo.owncloud.com"
...
+    with(account.javaClass.getDeclaredField("name")) {
+        isAccessible = true
+        set(account, name)
+    }
+    with(account.javaClass.getDeclaredField("type")) {
+        isAccessible = true
+        set(account, type)
+    }
+    return account
+}
```
Comment (JuancaG05) [inline owncloudTestUtil/src/main/java/com/owncloud/android/testutil/OCAccount.kt:44]: 🥇

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsPictureUploadsFragment.kt [hunk 3]
```diff
@@ -80,6 +82,10 @@ class SettingsPictureUploadsFragment : PreferenceFragmentCompat() {
         prefPictureUploadsOnWifi = findPreference(PREF__CAMERA_PICTURE_UPLOADS_WIFI_ONLY)
         prefPictureUploadsSourcePath = findPreference(PREF__CAMERA_PICTURE_UPLOADS_SOURCE)
         prefPictureUploadsBehaviour = findPreference(PREF__CAMERA_PICTURE_UPLOADS_BEHAVIOUR)
+        prefPictureUploadsAccount = findPreference<ListPreference>(PREF__CAMERA_PICTURE_UPLOADS_ACCOUNT_NAME)?.apply {
```
[the remark this answers is not in the document]
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsPictureUploadsFragment.kt:85]: Yes, it is. I need to set the entries dynamically, so I need to set ListPreference explicitly 👍

File: owncloudApp/src/main/res/xml/settings_video_uploads.xml (hunk 1 above)
Comment (abelgardep) [reply owncloudApp/src/main/res/xml/settings_video_uploads.xml:52]: Yes, it is needed if we do not want to show the cancel button in the dialog. So this is a glitch, good catch!

File: owncloudApp/src/main/java/com/owncloud/android/files/services/CameraUploadsHandler.java [hunk 4]
```diff
@@ -85,15 +85,14 @@ public void scheduleCameraUploadsSyncJob(Context context) {
...
-        extras.putString(Extras.EXTRA_ACCOUNT_NAME, mCameraUploadsConfig.getUploadAccountName());
-
         if (mCameraUploadsConfig.isEnabledForPictures()) {
             extras.putString(Extras.EXTRA_CAMERA_UPLOADS_PICTURES_PATH, mCameraUploadsConfig.
                     getUploadPathForPictures());
             extras.putString(Extras.EXTRA_CAMERA_UPLOADS_PICTURES_SOURCE_PATH, mCameraUploadsConfig.
                     getSourcePathPictures());
             extras.putInt(Extras.EXTRA_CAMERA_UPLOADS_PICTURES_BEHAVIOR_AFTER_UPLOAD, mCameraUploadsConfig.
                     getBehaviourAfterUploadPictures());
+            extras.putString(Extras.EXTRA_CAMERA_UPLOADS_PICTURES_ACCOUNT, mCameraUploadsConfig.getUploadAccountNameForPictures());
```
[the remark this answers is not in the document]
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/files/services/CameraUploadsHandler.java:95]: I think so, but I can keep the previous one. No problem from my side 👍

**Comment by abelgardep** [comment]
Changes applied, ready for a second-round @JuancaG05

**Comment by jesmrec** [comment]
Test plan has already passed successfully.

Focus was on the account uploads, not the camera uploads themselves. Camera uploads will be reviewed completely in the regression stage attached to the current version, just before releasing to the stores.

**Pull request merged**